lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -31,14 +31,17 @@ class KubernetesResource
     TIMEOUT_OVERRIDE_ANNOTATION = "kubernetes-deploy.shopify.io/timeout-override"
 
     class << self
-      def build(namespace:, context:, definition:, logger:, statsd_tags:)
+      def build(namespace:, context:, definition:, logger:, statsd_tags:, crds: [])
         opts = { namespace: namespace, context: context, definition: definition, logger: logger,
                  statsd_tags: statsd_tags }
         if definition["kind"].blank?
           raise InvalidTemplateError.new("Template missing 'Kind'", content: definition.to_yaml)
-        elsif KubernetesDeploy.const_defined?(definition["kind"])
+        elsif KubernetesDeploy.const_defined?(definition["kind"]) # Hardcoded CRs
           klass = KubernetesDeploy.const_get(definition["kind"])
           klass.new(**opts)
+        elsif crds.map(&:kind).include? definition["kind"]        # Dynamic CRs
+          crd = crds.find { |crd| crd.kind == definition["kind"] }
```

Comment:
nit: bugs me a little to see the same search performed twice, it seems to me the whole if would be better written as a series of guardclauses with explicit returns.